.github/workflows/pr.yml

```diff
@@ -0,0 +1,27 @@
+---
+name: Pull Request Workflow
+on:
+  pull_request:
+    branches:
+      - master
+jobs:
+  test:
```

Comment:
Do we need duplication of this job? If so, can't we write a template, and use it multiple times instead?

Reply:
Good point. But, reusing jobs is not as smooth as in GitLab. Especially, these jobs are not fully the same, and handling non-duplicate code increases complexity, which IMO isn't worth it.

Reply:
Links:
1. Struggles about supporting code reuse in GitHub Actions: [link](https://github.com/actions/starter-workflows/issues/245)
2. Current solution for action-level code reuse: [link](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action)

Reply:
Nice one, thanks for following up.